### PR TITLE
[Static Runtime] Enabled building Static Runtime tests and benchmarks in OSS CI

### PR DESCRIPTION
**Pull request by navahgar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #62226

Differential Revision: [D29923800](https://our.internmc.facebook.com/intern/diff/D29923800)

### CODE REVIEW

**Comment by navahgar** [comment]
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29923800).

**Comment by navahgar** [comment]
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29923800).

**Review by malfet — CHANGES_REQUESTED** [review]
It does not sound like a good idea to force building this benchmark for all opensource users by default. This could be enabled in CI by adding `export BUILD_STATIC_RUNTIME_BENCHMARK=ON` to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/build.sh for one of the supported configs, for example, similarly how torch_deploy is tested right now:
https://github.com/pytorch/pytorch/blob/e3be185069b3ae183c3575b8fe5ebe44f6728b62/.jenkins/pytorch/build.sh#L27-L30

**Comment by navahgar** [comment]
> It does not sound like a good idea to force building this benchmark for all opensource users by default. This could be enabled in CI by adding `export BUILD_STATIC_RUNTIME_BENCHMARK=ON` to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/build.sh for one of the supported configs, for example, similarly how torch_deploy is tested right now:
> https://github.com/pytorch/pytorch/blob/e3be185069b3ae183c3575b8fe5ebe44f6728b62/.jenkins/pytorch/build.sh#L27-L30

Thanks for that pointer. That is very helpful.

While I know that we don't want to enable this for `*mobile*` builds, I am not sure about the rest. Is there anyway to figure that out? Is there any place where I can find the full list of builds?

**Comment by navahgar** [comment]
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29923800).

**Comment by navahgar** [comment]
Also, one other related question: how can I ensure the tests built by setting this env variable, specifically `build/bin/static_runtime_test` runs every time in CI? @malfet

File: .jenkins/pytorch/build.sh [hunk 1]
```diff
@@ -210,6 +210,12 @@ if [[ "${BUILD_ENVIRONMENT}" == pytorch-linux-xenial-py3.6-gcc7-build || "${BUIL
   export USE_GLOO_WITH_OPENSSL=ON
 fi
 
+if [[ "${BUILD_ENVIRONMENT}" == pytorch-linux-xenial-py3* ]]; then
+  if [[ "${BUILD_ENVIRONMENT}" != *android* && "${BUILD_ENVIRONMENT}" != *ios* && "${BUILD_ENVIRONMENT}" != *mobile* && "${BUILD_ENVIRONMENT}" != *cuda* ]]; then
```
Comment (malfet) [inline .jenkins/pytorch/build.sh:214]: Android and cuda should be sufficient, shouldn't it, as `ios` builds could not run on linux
```suggestion
  if [[ "${BUILD_ENVIRONMENT}" != *android* && "${BUILD_ENVIRONMENT}" != *cuda* ]]; then
```

**Review by malfet — APPROVED** [review]

**Comment by navahgar** [comment]
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29923800).

**Comment by navahgar** [comment]
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29923800).

**Comment by navahgar** [comment]
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29923800).

**Comment by navahgar** [comment]
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29923800).

**Comment by navahgar** [comment]
@navahgar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29923800).

**Pull request closed without merging**